src/java/org/apache/cassandra/db/compression/CompressionDictionaryManager.java

```diff
@@ -370,11 +391,6 @@ private CompressionDictionaryTrainingConfig createTrainingConfig()
 
     private void storeDictionary(CompressionDictionary dictionary)
     {
-        if (!isEnabled)
```

Comment:
if we do not add to cache then who is actually releasing it? ... right ... nobody!